test/e2e/test-adchoices.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describes.endtoend(
+  'ad choices',
+  {
+    testUrl: 'http://localhost:8000/test/fixtures/e2e/amphtml-ads/text.html',
+    environments: 'amp4ads-preset',
+  },
+  env => {
+    let controller;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('interactions', async () => {
```

Comment:
I'd recommend splitting these interactions up into seperate tests.

```javascript
it('Information displayed when clicking info button', async () => {
  const infoDiv = await controller.findElement('#spv1');
  const infoBtn = await controller.findElement('#cbb');

  await expect(controller.getElementCssValue(infoDiv, 'top')).to.be.equal('-250px');
  await controller.click(infoBtn);
  await expect(controller.getElementCssValue(infoDiv, 'top')).to.be.equal('0px');
});
```

Reply:
sounds good idea. in this way the test description can be more informative.

Reply:
This specific test would not be severable. We click the info logo first, and then it shows a page with a button called "Why am I seeing this?".